Tests/AsyncHTTPClientTests/TransactionTests.swift

```diff
@@ -541,16 +541,26 @@ final class TransactionTests: XCTestCase {
 // tasks. Since we want to wait for things to happen in tests, we need to `async let`, which creates
 // implicit tasks. Therefore we need to wrap our iterator struct.
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-actor SharedIterator<Iterator: AsyncIteratorProtocol> {
-    private var iterator: Iterator
+actor SharedIterator<Wrapped: AsyncSequence> where Wrapped.Element: Sendable {
```

Comment:
`Wrapped.Element: Sendable` constraint necessary in Swift 5.8 because return types of actor isolated functions need to be `Sendable` [since `SE-0338`](https://github.com/apple/swift-evolution/blob/main/proposals/0338-clarify-execution-non-actor-async.md) and `Wrapped.Element` is the return type of `next()`.